pkg/ddl/executor.go

```diff
@@ -4515,6 +4506,28 @@ func GetName4AnonymousIndex(t table.Table, colName pmodel.CIStr, idxName pmodel.
 	return indexName
 }
 
+func checkCreateUniqueGlobalIndex(ec errctx.Context, tblInfo *model.TableInfo, indexName string, indexColumns []*model.IndexColumn, isUnique bool, isGlobal bool) error {
+	if isGlobal && tblInfo.GetPartitionInfo() == nil {
```

Comment:
Maybe use a helper variable to make it more readable, and use it for the whole function?
```suggestion
	pi := tblInfo.GetPartitionInfo()
	if isGlobal && pi == nil {
```